### ISSUE TITLE
Feature request: allow options at the end of the command line

### ISSUE THREAD

**Issue by premek** [ggreer/the_silver_searcher#1129]
Current: 
```
SYNOPSIS
       ag [options] pattern [path ...]

```
What would also be very nice:
```
       ag pattern [options] [path ...]
       ag pattern [path ...] [options]
```
Use case:
- `ag pattern`
- realise you want to search the filenames
- Actual: <kbd>Up</kbd>, <kbd>Home</kbd>, <kbd>Right</kbd>, <kbd>Right</kbd>, <kbd>Right</kbd>, type
- Nicer: <kbd>Up</kbd>, type
- 4 key strokes saved :)

**Comment by aswild** [comment]
If you're using a GNU libc, this behavior is already built in.  From the manpage for `getopt` and `getopt_long`:

>By  default, getopt() permutes the contents of argv as it scans, so that eventually all the nonoptions are at the end.

**Comment by premek** [comment]
Youre right, this works fine for all the options (e.g. `ag pattern --java`) but not for `-g` which I forgot to mention in my original description.
For `-g` the pattern is an argument of `-g` not the same as the global pattern.
What I wanted to do was:
```
ag pattern
ag pattern -g
```
